tests/functionTests/rpcTimeoutFuntionTest.txt

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 IBM Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##############################################################################
+# Default RPC timeout
+##############################################################################
+
+#START_TEST unary RPC with default timeout
+@@CMD@@ localhost:50051 examples.ScalarTypeRpcs incrementNumbers m_int32=3 m_int32=5
+/.* Received message:
+| m_double = 1.000000
+| m_float. = 1.000000
+| m_int32. = 6
+| m_int64. = 1
+| m_uint32 = 1 (0x00000001)
+| m_uint64 = 1 (0x0000000000000001)
+RPC succeeded :D
+#END_TEST
+
+#START_TEST unary RPC exeeding default timeout
+@@CMD@@ localhost:50051 examples.StatusHandling neverEndingRpc
+/.* Deadline Exceeded
+#END_TEST
+
+#START_TEST streaming RPC with infinite timeout
+@@CMD@@ localhost:50051 examples.StreamingRpcs bidirectionalStreamEchoNumberOrString :both=:str=High number=5::
+/.* Received message:
+| number = [NOT SET]
+| str... = [NOT SET]
+| both.. = {NumberAndString}
+| | number = 5
+| | str... = "High"
+RPC succeeded :D
+#END_TEST
+
+##############################################################################
+# Custom RPC timeout
+##############################################################################
+
+#START_TEST unary RPC with custom timeout
+@@CMD@@ --rpcTimeoutMilliseconds=500 localhost:50051 examples.NestedTypeRpcs echoNestedMaps
+Warning: no Fields found in parseTree for message 'NestedMaps'
+/.* Received message:
+| number_and_string..... = [NOT SET]
+| sub_message........... = [NOT SET]
+| simple_map_int[0/0]... = {}
+| simple_map_string[0/0] = {}
+RPC succeeded :D
+#END_TEST
+
+#START_TEST unary RPC exeeding custom timeout
+@@CMD@@ --rpcTimeoutMilliseconds=500 localhost:50051 examples.StatusHandling neverEndingRpc
+/.* Deadline Exceeded
+#END_TEST
+
+#START_TEST streaming RPC with custom timeout
+@@CMD@@ --rpcTimeoutMilliseconds=500 localhost:50051 examples.StreamingRpcs bidirectionalStreamEchoNumberOrString :number=5:
+/.* Received message:
+| number = 5
+| str... = [NOT SET]
+| both.. = [NOT SET]
+RPC succeeded :D
+#END_TEST
+
+#START_TEST streaming RPC exeeding custom timeout
+@@CMD@@ --rpcTimeoutMilliseconds=500 localhost:50051 examples.StreamingRpcs bidirectionalStreamInfiniteRpc
```

Comment:
there is AFAIK a test server RPC which is outputting the current time in a specified frequency. You could use this to more acurately test streamign timeouts in addition to never ending RPC